### PR TITLE
Update statediff to use fil_actor_interface

### LINKED ISSUE #2639: Use the actor interface in `forest_statediff` rather than depending directly on the individual actors
**Issue summary**

Right now, the forest_statediff tool depends directly on the versioned actors. It would be a big improvement if it used the actor interface to load the data without having to worry about versioning.

**Other information and links**

### PR DESCRIPTION
**Pull request by sudo-shashank**
## Summary of changes

Changes introduced in this pull request:
- Use `fil_actor_interface` in `statediff`

## Reference issue to close (if applicable)

Closes #2639 

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG](https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md) is up-to-date. All user-facing changes should be reflected in this document.

### CODE REVIEW

File: utils/statediff/src/lib.rs [hunk 1]
```diff
@@ -124,46 +120,41 @@ fn pp_actor_state(
...
-
-    // FIXME: Use the actor interface to load and pretty print the actor states.
-    //        Tracker: https://github.com/ChainSafe/forest/issues/1561
-    if let Ok(miner_state) = forest_ipld::from_ipld::<MinerState>(ipld.clone()) {
+    if let Ok(miner_state) = MinerState::load(bs, &state.into()) {
         write!(&mut buffer, "{miner_state:?}")?;
         return Ok(buffer);
     }
-    if let Ok(cron_state) = forest_ipld::from_ipld::<CronState>(ipld.clone()) {
+    if let Ok(cron_state) = CronState::load(bs, &state.into()) {
```
Comment (LesnyRumcajs) [inline utils/statediff/src/lib.rs:129]: @sudo-shashank Have you confirmed that this works locally? Sadly, we have no unit tests for now.

Comment (sudo-shashank) [reply utils/statediff/src/lib.rs:129]: Not yet

Comment (lemmih) [reply utils/statediff/src/lib.rs:129]: Add the missing actors and I will write unit tests.

File: utils/statediff/src/lib.rs [hunk 2]
```diff
@@ -11,15 +11,11 @@ use std::{
...
-use fil_actor_cron_v9::State as CronState;
-use fil_actor_init_v9::State as InitState;
-use fil_actor_market_v9::State as MarketState;
-use fil_actor_miner_v9::State as MinerState;
-use fil_actor_multisig_v9::State as MultiSigState;
-use fil_actor_power_v9::State as PowerState;
-use fil_actor_reward_v9::State as RewardState;
-use fil_actor_system_v9::State as SystemState;
+use fil_actor_interface::{
+    account::State as AccountState, cron::State as CronState, init::State as InitState,
```
Comment (LesnyRumcajs) [inline utils/statediff/src/lib.rs:15]: How do you specify the version now?

**Comment by LesnyRumcajs** [comment]
@sudo-shashank Have you confirmed that the tool still works after those changes?

**Comment by sudo-shashank** [comment]
@LesnyRumcajs yes

**Review by LesnyRumcajs — APPROVED** [review]

**Comment by lemmih** [comment]
This PR works. I'll add some tests and then approve it.

**Review by lemmih — APPROVED** [review]
Added a very basic test. Testing diffs would also be nice but that can be done later.

**Pull request merged**